src/Chat/MessageStore/CacheStore.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+declare(strict_types=1);
+
+namespace PhpLlm\LlmChain\Chat\MessageStore;
+
+use PhpLlm\LlmChain\Chat\MessageStoreInterface;
+use PhpLlm\LlmChain\Model\Message\MessageBag;
+use Psr\Cache\CacheItemPoolInterface;
+
+final readonly class CacheStore implements MessageStoreInterface
+{
+    public function __construct(
+        private CacheItemPoolInterface $cache,
+        private string $cacheKey = 'messages',
```

Comment:
using it in parallel would overwrite the messages of another conversation

Reply:
would you be more strict and enforce the key is injected?